ui/src/hooks/useTerminalConnection.ts

```diff
@@ -415,24 +385,22 @@ export const useTerminalConnection = (lunaId: Ref<string>, origin: Ref<string>)
     };
 
     terminal.onData((data: string) => {
-      let processedData = preprocessInput(data, terminalSettingsStore.getConfig);
+      const processedData = preprocessInput(data, terminalSettingsStore.getConfig);
 
       lastSendTime.value = new Date();
 
-      sendEventToLuna('KEYBOARDEVENT', '');
+      sendLunaEvent('KEYBOARDEVENT', '');
 
+      if (isClosed.value) {
+        return;
+      }
       socket.send(formatMessage(terminalId.value, FORMATTER_MESSAGE_TYPE.TERMINAL_DATA, processedData));
     });
-
-    terminal.attachCustomKeyEventHandler((e: KeyboardEvent) => {
-      return handleCustomKey(e, terminal, lunaId.value, origin.value);
-    });
   };
 
   return {
-    getShareUser,
     setShareCode,
-    terminalResizeEvent,
-    initializeSocketEvent
+    initializeSocketEvent,
+    eventBus,
   };
 };
```

Comment:
The only difference I spotted was this line:

`return clearInterval(pingInterval.value!);`

This seems to be a leftover of when the ping interval was set earlier. The variable `pingInterval`, which is defined before it's declared in the function declaration, would usually have been reset outside of these functions and not stored in an object scope like that.

As such, this can be removed without impacting other parts of the app logic or causing bugs.

Would you like me to review any others?